### PR TITLE
Ccip rev share

### PR DESCRIPTION
**Pull request by Gorshtak**

### CODE REVIEW

File: src/queries/token/useGlobalStakingDataQuery.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            const stakingData: GlobalStakingData = {
+                totalStakedAmount: 0,
+                thalesApy: 0,
+                feeApy: 0,
+                baseRewards: 0,
+                extraRewards: 0,
+            };
+            try {
+                // Thales staked - Base
+                const baseAnkrProvider = new ethers.providers.JsonRpcProvider(
```
Comment (vpoklopic) [inline src/queries/token/useGlobalStakingDataQuery.ts:23]: Use chainnodes here

File: src/queries/token/useGlobalStakingDataQuery.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                    ((bigNumberFormatter(stakedAmount) + bigNumberFormatter(escrowedAmount)) * thalesTokenPrice);
+
+                const thalesRewardsAPY =
+                    ((bigNumberFormatter(baseRewardsPool) + bigNumberFormatter(bonusRewardsPool)) * 52 * 100) /
+                    (bigNumberFormatter(stakedAmount) + bigNumberFormatter(escrowedAmount));
+
+                stakingData.feeApy = aprToApy(stakingData.feeApy);
+                stakingData.thalesApy = aprToApy(thalesRewardsAPY);
+
+                stakingData.feeApy = Number((Math.round(stakingData.feeApy * 100) / 100).toFixed(2));
```
Comment (vpoklopic) [inline src/queries/token/useGlobalStakingDataQuery.ts:65]: Why * 100 and / 100?

File: src/queries/token/useGlobalStakingDataQuery.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                    CCIPCollectorContract.calculatedEscrowedAmountForPeriod(Number(period)),
+                    CCIPCollectorContract.calculatedRevenueForPeriod(Number(period)),
+                    CCIPCollectorContract.baseRewardsPerPeriod(),
+                    CCIPCollectorContract.extraRewardsPerPeriod(),
+                ]);
+
+                const revShare = period <= 3 ? 30000 : bigNumberFormatter(calculatedRevenueForPeriod);
+                const thalesTokenPrice = Number(await price.text());
+
+                stakingData.feeApy =
```
Comment (vpoklopic) [inline src/queries/token/useGlobalStakingDataQuery.ts:54]: please name it right, this is APR

File: src/queries/token/useGlobalStakingDataQuery.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                    (revShare * 52 * 100) /
+                    ((bigNumberFormatter(stakedAmount) + bigNumberFormatter(escrowedAmount)) * thalesTokenPrice);
+
+                const thalesRewardsAPY =
```
Comment (vpoklopic) [inline src/queries/token/useGlobalStakingDataQuery.ts:58]: same here

**Review by vpoklopic — CHANGES_REQUESTED** [review]

File: src/queries/token/useGlobalStakingDataQuery.ts (hunk 2 above)
Comment (Gorshtak) [reply src/queries/token/useGlobalStakingDataQuery.ts:65]: Stack overflow answer to round up to 2 decimals with all edge cases covered.

**Review by vpoklopic — APPROVED** [review]

**Pull request merged**